iox_data_generator/src/tag_set.rs

```diff
@@ -461,8 +460,8 @@ impl std::fmt::Display for TagSet {
 
 #[derive(Debug, PartialEq, PartialOrd)]
 pub struct TagPair {
-    key: Arc<String>,
-    value: Arc<String>,
+    pub key: Arc<String>,
+    pub value: Arc<String>,
```

Comment:
Because `String`s are a pointer to bytes on the heap, and `Arc`s are a pointer to the heap, this is adding an extra layer of indirection. This can be:

```suggestion
    pub key: Arc<str>,
    pub value: Arc<str>,
```

and where you construct `TagPair`s, if you have a `String` you can do `string.as_str().into()` (deref coercion isn't being as helpful as I wish it would be here) and if you have a `&str` you can do `string_slice.into()` to get `Arc<str>`s.

I don't know if or how much this is impacting the performance, but making the change certainly can't *hurt*.